builder/main.py

```
@@ -133,6 +150,275 @@ def fetch_fs_size(env):
+        workarea_size = sector_size * 2
+        
+        ret = pyf_mkfs(
+            base_partition.pname, 
+            n_fat=2, 
+            align=0,
+            n_root=512,
+            au_size=0,
+            workarea_size=workarea_size
+        )
+        if ret != FR_OK:
+            raise Exception(f"Failed to format filesystem: error code {ret}")
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)